framework/sandstone_tests.cpp

```diff
@@ -79,51 +79,42 @@ SandstoneTestSet::SandstoneTestSet(struct test_set_cfg cfg, unsigned int flags)
         struct test_cfg_info ti;
         ti.status = test_cfg_info::enabled;
         ti.test = &test;
-        test_map[test.id] = ti;
         test_set.push_back(&test);
     }
 };
 
-struct test_cfg_info SandstoneTestSet::enable(struct test *t) {
-    struct test_cfg_info &ti = test_map[t->id];
+struct test_cfg_info SandstoneTestSet::append(test_cfg_info t)
+{
+    struct test_cfg_info &ti = test_set.emplace_back(t);
+    // ensure it's enabled
     ti.status = test_cfg_info::enabled;
-    ti.test = t;
     return ti;
 }
 
 std::vector<struct test_cfg_info> SandstoneTestSet::enable(const char *name) {
     std::vector<struct test_cfg_info> res;
     std::vector<struct test *> tests = lookup(name);
     for (auto t : tests) {
-        struct test_cfg_info ti = enable(t);
+        struct test_cfg_info ti = append(t);
         res.push_back(ti);
-        test_set.push_back(t);
     }
     return res;
 }
 
-struct test_cfg_info SandstoneTestSet::disable(struct test *t) {
-    struct test_cfg_info &ti = test_map[t->id];
-    ti.status = test_cfg_info::disabled;
-    ti.test = t;
-    return ti;
+void SandstoneTestSet::remove(const struct test *test)
+{
+    auto it = std::remove_if(test_set.begin(), test_set.end(), [&](const test_cfg_info &ti) {
+        return ti.test == test;
+    });
+    test_set.erase(it, test_set.end());
 }
 
-std::vector<struct test_cfg_info> SandstoneTestSet::disable(const char *name) {
-    std::vector<struct test_cfg_info> res;
+void SandstoneTestSet::disable(const char *name)
```

Comment:
Do not remove the return value. I know it is not being used now, but the plan was to improve diagnostics to warn that names didn't match any.

Reply:
For that you don't need a list of tests that did match. You just need a count, because `name` is what you want to warn on.

Changing to return the count of disabled tests.